### ISSUE TITLE
Upload files not working with NodeBB 1.8.1

### ISSUE THREAD

**Issue by Silvia-t** [NodeBB/nodebb-plugin-composer-redactor#71]
I've recently upgraded NodeBB to version 1.8.1 and uploading files with the composer redactor doesn't work.

It shows the blue line at the top like it's uploading, but that's it.

Here's a screenshot showing that all the compatibility checks are passed.

![image](https://user-images.githubusercontent.com/23095224/37901752-31c3cdd8-30ea-11e8-8e4a-d49f11ce0c48.png)

I've had to revert back to the default composer :(

**Comment by beerpromax** [comment]
Same problem as the uploading processes ends, there's no image showing in the text field. My nodebb version is 1.8.2 :/ have to revert

**Comment by julianlam** [comment]
Please let me know your redactor plugin version... is it v3.3.0 or higher?

**Comment by Silvia-t** [comment]
I upgraded to NodeBB v1.9.3 the other day and now the redactor (version 3.2.0, which seems to be the most recent one?) seems to be working fine :)

**Comment by Silvia-t** [comment]
Actually, it's NOT working fine u.u

It did when I tested it earlier today but now it's not working again (which seems quite random)...

**Comment by julianlam** [comment]
Yes, version v3.2.0 addresses a file upload issue. Are you running in a subfolder?

**Comment by Silvia-t** [comment]
Hi @julianlam, I'm not too sure I understand about the subfolder? Do you mean NodeBB?

I've tried it again and it looked like I could upload anything when creating a new post, but on replying it wouldn't upload anything :(

**Comment by julianlam** [comment]
@Silvia-t I mean, are you accessing NodeBB via a subfolder... e.g. http://example.com/forum ?

**Comment by Silvia-t** [comment]
No, NodeBB is hosted in our server so I'm accessing it through http://forum.ourdomain.co.uk/

**Comment by julianlam** [comment]
It seems I can upload regular files, but it does what you say when I try to upload an **image**... uploading images is a different button. Does this work now?

I agree that uploading an image via the file upload dialog should also work, so that's probably a bug. Just wanted to make sure we're all on the same page now

**Comment by julianlam** [comment]
![selection_428](https://user-images.githubusercontent.com/923011/41860102-bf5dd6e8-786b-11e8-96f8-c4b340bc3257.png)

Use the former to upload images, use the latter for files.

**Comment by Silvia-t** [comment]
Hi @julianlam, yes, I was aware that uploading images doesn't work through the file upload (although I agree with you in the sense that it should work!).

Yesterday I could upload regular files, but only if it was in the first post. Nothing would attach in a reply.

Anyway, I'm off for over a week now so excuse my silence in the meantime! :P

**Comment by dongcai** [comment]
Not very sure why. But installing 'sharp' fixes my issue, like below:
`npm install --build-from-source sharp`
BTW, I am running NodeBB 1.12.1
